lib/ex_azure_key_vault.ex

```diff
@@ -91,6 +194,17 @@ defmodule ExAzureKeyVault.Client do
       }
 
   """
+  @spec connect!() :: Client.t() | {:error, any}
```

Comment:
This seems out of scope as well. Also, although I understand the motivation for the `!` function, the current `connect()` already raises exceptions.

Please open a PR or issue with this suggestion if you want to discuss more.